specification/marketplacecatalog/resource-manager/Keys/stable/2023-01-01-preview/discoverykeys.json

```diff
@@ -0,0 +1,255 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Marketplace Discovery API Keys Management",
+    "version": "2023-01-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Marketplace/discovery/keys/create": {
```

Comment:
>discovery [](http://example.com/codeflow?start=69&length=9)

"discovery" doesn't seem to be needed here, it can be simply modeled as "/keys/create" if you want to create a key at the sub scope. As a general rule, after the namespace you'll see a resource type (named in plural), in your case you have 2 resource types, "discovery" and "keys" which is invalid. See our [RPC](https://github.com/Azure/azure-resource-manager-rpc/blob/c71ccbd6a8adf4a47978e46bda1e7d405cb19b80/v1.0/proxy-api-reference.md#subscription-wide-reads-and-actions) for reference